src/plot/series/custom-svg-series.js

```diff
@@ -0,0 +1,159 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import AbstractSeries from './abstract-series';
+import Animation from 'animation';
+import {ANIMATED_SERIES_PROPS} from 'utils/series-utils';
+
+const predefinedClassName = 'rv-xy-plot__series rv-xy-plot__series--custom-svg-wrapper';
+
+const DEFAULT_STYLE = {
+  stroke: 'blue',
+  fill: 'blue'
+};
+
+function predefinedComponents(type, size = 2, style = DEFAULT_STYLE) {
+  switch (type) {
+  case 'diamond':
+    return (<polygon
+      style={style}
+      points={`0 0 ${size / 2} ${size / 2} 0 ${size} ${-size / 2} ${size / 2} 0 0`} />);
+  case 'star':
+    const starPoints = [...new Array(5)].map((c, index) => {
+      const angle = index / 5 * Math.PI * 2;
+      const innerAngle = angle + Math.PI / 10;
+      const outerAngle = angle - Math.PI / 10;
+      // ratio of inner polygon to outer polgyon
+      const innerRadius = size / 2.61;
+      return `
+        ${Math.cos(outerAngle) * size} ${Math.sin(outerAngle) * size}
+        ${Math.cos(innerAngle) * innerRadius} ${Math.sin(innerAngle) * innerRadius}
+      `;
+    }).join(' ');
+    return (<polygon
+        points={starPoints}
+        x="0" y="0" height={size} width={size} style={style}/>);
+  case 'square':
+    return (<rect x={`${-size / 2}`} y={`${-size / 2}`} height={size} width={size} style={style}/>);
+  default:
+  case 'circle':
+    return (<circle cx="0" cy="0" r={size / 2} style={style}/>);
+  }
+}
+
+function getInnerComponent({customComponent, positionInPixels, defaultType, positionFunctions}) {
+  const {size, style} = customComponent;
+  const innerComponent = customComponent.customComponent;
+  if (!innerComponent && typeof defaultType === 'string') {
+    return predefinedComponents(defaultType, size, style);
+  }
+  // if default component is a function
+  if (!innerComponent) {
+    return innerComponent(defaultType, positionInPixels);
+  }
+  if (typeof innerComponent === 'string') {
+    return predefinedComponents(innerComponent || defaultType, size, style);
+  }
+  // if inner component is a function
+  return innerComponent(customComponent, positionInPixels);
+}
+
+class CustomSVGSeries extends AbstractSeries {
+  render() {
+    const {
+      animation,
+      className,
+      customComponent,
+      data,
+      innerHeight,
+      innerWidth,
+      marginLeft,
+      marginTop,
+      style
+    } = this.props;
+
+    if (!data || !innerWidth || !innerHeight) {
+      return null;
+    }
+
+    if (animation) {
+      return (
+        <Animation {...this.props} animatedProps={ANIMATED_SERIES_PROPS}>
+          <CustomSVGSeries {...this.props} animation={null}/>
```

Comment:
animation is more than just a boolean! it can also be a string or an object. I'm 99% sure that if i put animation={false} as jsx animation will still be read as true (while if i were to spread an object of props that contain animation: false, that would be read correctly)